core/controllers/acl_decorators.py

```diff
@@ -577,6 +577,46 @@ def test_super_admin(self, **kwargs):
     return test_super_admin
 
 
+def can_access_contributor_dashboard_admin_page(handler):
+    """Decorator that checks if the current user is a contributor dashboard
+    admin.
+
+    Args:
+        handler: function. The function to be decorated.
+
+    Returns:
+        function. The newly decorated function that now also checks if the user
+        is a super admin.
+    """
+
+    def test_contributor_dashboard_admin(self, **kwargs):
+        """Checks if the user is logged in and is a contributor dashboard admin.
+
+        Args:
+            **kwargs: *. Keyword arguments.
+
+        Returns:
+            *. The return value of the decorated function.
+
+        Raises:
+            NotLoggedInException. The user is not logged in.
+            UnauthorizedUserException. The user is not a contributor dashboard
+                admin of the application.
+        """
+        if not self.user_id:
+            raise self.NotLoggedInException
+
+        if role_services.ACTION_MANAGE_CONTRIBUTORS in self.user.actions:
```

Comment:
Can you describe, in a bullet-point list, all the specific actions that the contributor dashboard admin can do?

For this particular case it sounds like "access contributor dashboard admin page", similar to what we did for release coordinator. But you might also want to break other actions down as well (e.g. "manage contributor dashboard roles").

Reply:
We allow following actions:
- Access contributor dashabord page.
- Add/remove reviewers for suggestions.
- Allow selected users to submit question suggestions.

Note: Currently the last two actions are performed by one handler so a combined action can be manage_community _contributions i.e, community contributions depends on author/reviewer and user with this role manages contributions by specificing who can craete suggestion who can review suggestions.

Reply:
Sounds good. Yes, let's have "access contributor dashboard admin page" and "manage contributor dashboard roles". I think the latter is better than "manage community contributions" because the thing that's more specifically being managed here is the roles and not the contributions in general.